packages/dev-server/src/HermesBundler.ts

```diff
@@ -9,17 +10,21 @@ import resolveFrom from 'resolve-from';
 
 export async function shouldBuildHermesBundleAsync(
   projectRoot: string,
+  target: ProjectTarget,
   platform: Platform
 ): Promise<boolean> {
+  if (target === 'managed') {
+    const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
+    return platform === 'android' && exp.android?.jsEngine === 'hermes';
+  }
```

Comment:
Once tried but later feel the original code might be clear after adding ios support.
Sorry for editing your suggestion code b/c trying to commit the suggestion before.

```ts
  if (target === 'managed') {
    const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
    switch (platform) {
      case 'android':
        return exp.android?.jsEngine === 'hermes';
      case 'ios':
        return exp.ios?.jsEngine === 'hermes';
      default:
        return false;
    }
  }

  // bare workflow

  if (platform === 'ios') {
    // Read settings somewhere or maybe directly from Podfile
    if (fs.existsSync(...)) {
      const settings = parseIosSettings(...); 
      return settings['jsEngine'] === 'hermes';
    }
  }

  if (platform === 'android') {
    const gradlePropertiesPath = path.join(projectRoot, 'android', 'gradle.properties');
    if (fs.existsSync(gradlePropertiesPath)) {
      const properties = parseGradleProperties(await fs.readFile(gradlePropertiesPath, 'utf8'));
      return properties['expo.jsEngine'] === 'hermes';
    }
  }
  return false;
```

v.s.

```ts
  if (platform === 'web') {
    return false;
  }

  if (platform === 'ios') {
    if (target === 'managed') {
      const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
      return exp.ios?.jsEngine === 'hermes';
    }

    // Read settings somewhere or maybe directly from Podfile
    if (fs.existsSync(...)) {
      const settings = parseIosSettings(...); 
      return settings['jsEngine'] === 'hermes';
    }
    return false;
  }

  // platform === 'android'
  if (target === 'managed') {
    const { exp } = getConfig(projectRoot, { skipSDKVersionRequirement: true });
    return exp.android?.jsEngine === 'hermes';
  }

  const gradlePropertiesPath = path.join(projectRoot, 'android', 'gradle.properties');
  if (fs.existsSync(gradlePropertiesPath)) {
    const properties = parseGradleProperties(await fs.readFile(gradlePropertiesPath, 'utf8'));
    return properties['expo.jsEngine'] === 'hermes';
  }

  return false;
```

Reply:
both of those alternatives seem nice to me. The one suggestion I would have would be to change the comments, `// bare workflow`, `// platform === 'android'` into `if` or `switch` clauses. 

This would keep the code symmetric between bare/managed and ios/android and also demarcate the relevant code block.

Reply:
make sense. let me update the code accordingly.